static/app/views/projectInstall/issueAlertNotificationOptions.tsx

```diff
@@ -0,0 +1,277 @@
+  ]);
+  const [provider, setProvider] = useState<string | undefined>(undefined);
+  const [integration, setIntegration] = useState<OrganizationIntegration | undefined>(
+    undefined
+  );
+  const [channel, setChannel] = useState<string | undefined>(undefined);
+
+  const api = useApi();
+  const organization = useOrganization();
+
+  type Props = {
+    actionMatch: string | undefined;
+    conditions: {id: string; interval: string; value: string}[] | undefined;
+    frequency: number | undefined;
+    name: string | undefined;
+    projectSlug: string;
+    shouldCreateRule: boolean | undefined;
+  };
+
+  const createNotificationAction = useCallback(
+    ({
+      shouldCreateRule,
+      projectSlug,
+      name,
+      conditions,
+      actionMatch,
+      frequency,
+    }: Props) => {
+      const isCreatingIntegrationNotification = actions.find(
+        action => action === MultipleCheckboxOptions.INTEGRATION
+      );
+      if (
+        !organization.features.includes(
+          'messaging-integration-onboarding-project-creation'
+        ) ||
+        !shouldCreateRule ||
+        !isCreatingIntegrationNotification
+      ) {
+        return undefined;
+      }
+
+      let integrationAction;
+      switch (provider) {
+        case 'slack':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.SLACK,
+              workspace: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        case 'discord':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.DISCORD,
+              server: integration?.id,
+              channel_id: channel,
+            },
+          ];
+          break;
+        case 'msteams':
+          integrationAction = [
+            {
+              id: IssueAlertActionType.MS_TEAMS,
+              team: integration?.id,
+              channel: channel,
+            },
+          ];
+          break;
+        default:
+          return undefined;
+      }
+
+      return api.requestPromise(`/projects/${organization.slug}/${projectSlug}/rules/`, {
+        method: 'POST',
+        data: {
+          name,
+          conditions,
+          actions: integrationAction,
+          actionMatch,
+          frequency,
+        },
+      });
+    },
+    [
+      actions,
+      api,
+      provider,
+      integration,
+      channel,
+      organization.features,
+      organization.slug,
+    ]
+  );
+
+  return {
+    createNotificationAction,
+    notificationProps: {
+      actions,
+      provider,
+      integration,
+      channel,
+      setActions,
+      setProvider,
+      setIntegration,
+      setChannel,
+    },
+  };
+}
+
+export default function IssueAlertNotificationOptions(
+  notificationProps: IssueAlertNotificationProps
+) {
+  const organization = useOrganization();
+  const {actions, provider, setActions, setIntegration, setProvider} = notificationProps;
```

Comment:
```suggestion
export default function IssueAlertNotificationOptions(
  {actions, provider, setActions, setIntegration, setProvider}: IssueAlertNotificationProps
) {
  const organization = useOrganization();
```

Reply:
Ah I wrote it this way bc `notificationProps` still needs to be passed to `MessagingIntegrationAlertRule` [here](https://github.com/getsentry/sentry/blob/mia/proj-creation/fe/static/app/views/projectInstall/issueAlertNotificationOptions.tsx#L253)

Reply:
Ah got it 👍